flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -167,6 +167,13 @@ public int releaseMemory() {
 	}
 
 	@Override
+	public boolean isReleased() {
```

Comment:
True, but a different thread than the consumer may release the partition and update the flag. I can make the flag `volatile`.